app/helpers/provisioning_templates_helper.rb

```diff
@@ -67,6 +67,11 @@ def permitted_actions(template)
     end
   end
 
+  def pxe_with_building_hosts?(template)
+    kinds = ["PXELinux", "PXEGrub", "iPXE"]
+    template.operatingsystems.map(&:hosts).flatten.any?(&:build) && !template.template_kind.nil? && kinds.include?(template.template_kind.name)
```

Comment:
We should check the performance of this host lookup query to check it's not performing a huge number of SQL queries.  We should probably also use `where(:build => true)` to filter in the database.